src/main/scala/org/apache/spark/mllib/linalg/distributed/BetterBlockMatrix.scala

```diff
@@ -0,0 +1,188 @@
+package org.apache.spark.mllib.linalg.distributed
```

Comment:
If you're in this package, can't you just reference GridPartitioner directly instead of using the Sneaky business?  (Not that throwing stuff into someone else's package isn't sneaky.)